updating tracker docs for 3.5

russorat opened the pull request:
Passes Quick Build: http://builds.cask.co/browse/CDAP-DQB101-6

Commit:
updating tracker docs for 3.5

sagarkapare commented:
Please cherry-pick the commits to release/3.5 and release/3.6 once it is merged.

Commit:
adding audit metrics endpoints

awholegunch commented on cdap-docs/tracker-manual/source/index.rst:
```diff
@@ -18,46 +18,53 @@ Cask Tracker ("Tracker") is a self-service CDAP extension that automatically cap
-describes, explains, locates, and makes it easier to retrieve, use, and manage datasets.
+It allows you to perform impact and root-cause analysis, delivers an audit-trail for
+auditability and compliance, and allows you to preview data. Tracker furnishes access to
+structured information that describes, explains, locates, and makes it easier to
```
Could this sentence become:

```
Tracker furnishes access to
structured information that describes, explains, and locates data, making it easier to
retrieve, use, and manage datasets.
```

awholegunch commented on cdap-docs/tracker-manual/source/index.rst:
```diff
@@ -18,46 +18,53 @@ Cask Tracker ("Tracker") is a self-service CDAP extension that automatically cap
-Tracker also allows for the storage of metadata where it can be accessed and indexed. This
-allows it to be is easily searched and provides consistent, high-quality metadata.
+Tracker also allows users to update metadata for datasets and streams. Users can add,
+remove, and update tags or user properties directly in the UI. We also allow users to set
```
update tags **and** user properties directly in the UI. **It allows** users to set

awholegunch commented on cdap-docs/tracker-manual/source/index.rst:
```diff
@@ -18,46 +18,53 @@ Cask Tracker ("Tracker") is a self-service CDAP extension that automatically cap
 <metadata-lineage-lineage>` of an entity. A lineage shows |---| for a specified time range
 |---| all data access of the entity, and details of where that access originated from.
 
+Tracker captures activity metrics for datasets as well. Users can see the datasets that are
```
```
Tracker also captures activity metrics for datasets. You can see the datasets that are
being used the most and view usage metrics for each dataset. This allows teams to easily
determine the appropriate dataset to use for an analysis. The Tracker Meter rates each 
dataset on a scale that shows how active a dataset is in the system.

```

awholegunch commented on cdap-docs/tracker-manual/source/index.rst:
```diff
@@ -18,46 +18,53 @@ Cask Tracker ("Tracker") is a self-service CDAP extension that automatically cap
+- Provides viewing and updating complete metadata on datasets, enabling traceability to resolve
   data issues and to improve data quality.
 
+- Collects usage metrics about datasets so that you know which datasets are being used the most.
```
which datasets are being used most often.

awholegunch commented on cdap-docs/tracker-manual/source/index.rst:
```diff
@@ -18,46 +18,53 @@ Cask Tracker ("Tracker") is a self-service CDAP extension that automatically cap
 
+- Collects usage metrics about datasets so that you know which datasets are being used the most.
+
+- Provides the ability to designate tags as preferred so that teams can easily find and tag datasets.
```
Should this be:

```
Provides the ability to designate certain tags as "preferred" so that teams can easily find and tag datasets.
```

awholegunch commented on cdap-docs/tracker-manual/source/index.rst:
```diff
@@ -76,15 +83,22 @@ three billion records <http://customers.cask.co/rs/882-OYR-915/images/tracker-ca
 ===================
-The Cask Tracker application consists of an application in CDAP with two programs and two datasets:
+The Cask Tracker application consists of an application in CDAP with two programs and
+four datasets:
```
What about instead:

```
Cask Tracker consists of an application in CDAP with two programs and four datasets:
```

awholegunch commented on cdap-docs/tracker-manual/source/index.rst:
```diff
@@ -76,15 +83,22 @@ three billion records <http://customers.cask.co/rs/882-OYR-915/images/tracker-ca
+- ``_timeSinceTable`` a custom dataset used to store the last time a specific audit
+  message was received
+
+The Tracker UI is shipped with CDAP, started automatically in standalone as part of the
```
...started automatically in standalone CDAP as...

awholegunch commented on cdap-docs/tracker-manual/source/index.rst:
```diff
@@ -76,15 +83,22 @@ three billion records <http://customers.cask.co/rs/882-OYR-915/images/tracker-ca
+- ``_auditMetrics`` a custom cube dataset used to collect metrics about datasets
+- ``_auditTagsTable`` a custom dataset used to store preferred tags
+- ``_timeSinceTable`` a custom dataset used to store the last time a specific audit
+  message was received
```
```
- ``TrackerService``: Service exposing the Tracker API endpoints
- ``AuditLogFlow``: Flow that subscribes to Kafka audit messages and stores them in the
  ``_auditLog`` dataset
- ``_auditLog``: Custom dataset for storing audit messages
- ``_kafkaOffset``: Key-value table for storing Kafka offsets
- ``_auditMetrics``: Custom cube dataset for collecting dataset metrics
- ``_auditTagsTable``: Custom dataset for storing preferred tags
- ``_timeSinceTable``: Custom dataset for storing the last time a specific audit
  message was received
```

awholegunch commented on cdap-docs/tracker-manual/source/index.rst:
```diff
@@ -133,8 +147,8 @@ To enable Tracker, go to the **Tracker UI** at
 (Distributed CDAP) http://host:dashboard-bind-port/ns/default/tracker/home 
 and press the ``"Enable Tracker"`` button to deploy and start Tracker.
```
Is this still required? In 3.5.0, it was already running when I went to `http://localhost:9999/ns/default/tracker/home`

russorat replied:
in standalone it is automatically enabled. in distributed, you have to manually enable it.

awholegunch commented on cdap-docs/tracker-manual/source/index.rst:
```diff
@@ -206,12 +222,10 @@ Disabling and Removing Tracker
 
-- stop the Tracker flow
-- stop the Tracker service
+- stop all _Tracker programs
```
```
- stop all ``_Tracker`` programs
```

awholegunch commented on cdap-docs/tracker-manual/source/index.rst:
```diff
@@ -246,8 +260,8 @@ The results would appear similar to this:
-used is shown in the upper-left, and the results show both the datasets found with
+In this example, Tracker has found a single dataset that satisfied the condition. The search
+used is shown in the upper-left, and the results show the dataset found with
 information and links for each.
```
You have changed the text, but the image still shows two datasets returned. Plus, though you changed the text to speak of a single dataset, the ending ("links for each") is plural.

Should you go back to the original text?

russorat replied:
fixed.

awholegunch commented on cdap-docs/tracker-manual/source/index.rst:
```diff
@@ -259,8 +273,8 @@ Note that the *entities* and *metadata* filters have an ``and`` relationship; at
 
-**On the right side** is a sortable list of results. It is sortable by *Create Date* or the entity
-ID (name), either *A-Z* (alphabetical ascending), or *Z-A* (alphabetical descending).
+**On the right side** is a sortable list of results. It is sortable by *Create Date*, the entity
```
```
It is sortable by one of the *Create Date*,
```

awholegunch commented on cdap-docs/tracker-manual/source/index.rst:
```diff
@@ -273,8 +287,8 @@ real-time pipelines.
-entity. Details are provided on three separate tabs: *Metadata*, *Lineage*, and *Audit
-Log*.
+entity. Details are provided on the separate tabs: *Metadata*, *Lineage*, *Audit Log*,
+*Preview*, and *Usage*.
```
Should that not be (removing _Preview_ and editing):

```
Details are provided on the tabs *Metadata*, *Lineage*, *Audit Log*, and *Usage*.
```

russorat replied:
this is for data preview, not pipeline preview.

awholegunch replied:
Yes, I understand now that Preview should show up, if appropriate; what about:

```
Details are provided on the tabs *Metadata*, *Lineage*, *Audit Log*, *Preview*
(included if the dataset is explorable), and *Usage*.
```

awholegunch commented on cdap-docs/tracker-manual/source/index.rst:
```diff
@@ -325,6 +344,57 @@ specified, in addition to common time ranges (two weeks to one year).
   :align: center
   :class: bordered-image
 
+**Preview**
```
This should be removed, as _Preview_ was not included.

awholegunch commented on cdap-docs/tracker-manual/source/index.rst:
```diff
@@ -325,6 +344,57 @@ specified, in addition to common time ranges (two weeks to one year).
 
+**Preview**
+
+The *Preview* tab shows a preview for the dataset. It is available for all datasets that are
```
```
The *Preview* tab (if included) shows a preview for the dataset.
```

awholegunch commented on cdap-docs/tracker-manual/source/index.rst:
```diff
@@ -325,6 +344,57 @@ specified, in addition to common time ranges (two weeks to one year).
+**Preview**
+
+The *Preview* tab shows a preview for the dataset. It is available for all datasets that are
+explorable. You can scroll for up to 500 records. For additional analysis, you can leverage the
```
```
For additional analysis, use the *Jump* menu to go into CDAP and explore the dataset using a custom query.
```

awholegunch commented on cdap-docs/tracker-manual/source/index.rst:
```diff
@@ -325,6 +344,57 @@ specified, in addition to common time ranges (two weeks to one year).
+
+The *Tags* tab at the top of the page allows you to enter a common set of preferred terms to use when
+adding tags to datasets. Preferred tags show up first when adding tags, and will guide your team to
+use the same terminology. Any Preferred Tag that has not been attached to any datasets can be deleted
```
Any preferred tag that has not been attached to any entities

awholegunch commented on cdap-docs/tracker-manual/source/index.rst:
```diff
@@ -325,6 +344,57 @@ specified, in addition to common time ranges (two weeks to one year).
+The *Tags* tab at the top of the page allows you to enter a common set of preferred terms to use when
+adding tags to datasets. Preferred tags show up first when adding tags, and will guide your team to
+use the same terminology. Any Preferred Tag that has not been attached to any datasets can be deleted
+by clicking the red trashcan icon. If a Preferred tag ahs been added to an entity, you cannot delete it,
```
If a preferred tag has been added to an entity,

awholegunch commented on cdap-docs/tracker-manual/source/index.rst:
```diff
@@ -325,6 +344,57 @@ specified, in addition to common time ranges (two weeks to one year).
+Adding preferred tags is easy. Just click the **Add Preferred Tags** button and use the UI to add a
+list of terms you would like to be preferred. If the tag already exists in CDAP, it will be promoted
+to a preferred tag automatically. If it is a new tag in CDAP, it will still show up in the Preferred
+Tags list.
```
```
To add preferred tags, click the **Add Preferred Tags** button and use the UI to add or import a
list of tags that you would like to be "preferred". If the tag already exists in CDAP, 
it will be promoted from being a user tag to being a preferred tag. If it is a new tag
in CDAP, it will be added in the *Preferred Tags* list.
```

awholegunch commented on cdap-docs/tracker-manual/source/index.rst:
```diff
@@ -346,12 +416,16 @@ Details on completing this form are described in CDAP's documentation on
-dataset, stream, or stream view, submit an HTTP GET request::
+All Tracker features are also available via RESTful endpoints. It supports searching of
+the *_auditLog* dataset, managing preferred tags, and querying metrics, all through a set
+of HTTP RESTful APIs. (See the :ref:`Reference Manual: HTTP RESTful API
```
```
the *_auditLog* dataset, managing preferred tags, and querying metrics, through a set
of HTTP RESTful APIs.
```

awholegunch commented on cdap-docs/tracker-manual/source/index.rst:
```diff
@@ -483,3 +557,650 @@ Results (reformatted for display)::
+     * - Parameter
+       - Description
+     * - ``type`` *(optional)*
+       - The type, either ``user`` or ``preferred`` of tag to return
```
The type of tag to return, either `user` or `preferred`

Commit:
fixing comments and formatting

Commit:
fixing line breaks for urls

Commit:
fixing list issues

awholegunch commented on cdap-docs/tracker-manual/source/index.rst:
```diff
@@ -346,13 +420,16 @@ Details on completing this form are described in CDAP's documentation on
-the :ref:`Reference Manual: HTTP RESTful API <http-restful-api-introduction>` for details
-on the conventions used for this API.) To search for audit log entries for a particular
-dataset, stream, or stream view, submit an HTTP GET request::
+All Tracker features are also available via RESTful endpoints. It supports searching of
```
```
All Tracker features are available via HTTP RESTful endpoints.
```

Commit:
fixing more list issues

Commit:
fixing additional comments

russorat commented:
@awholegunch addressed comments. build is failing from hydrator docs issues, not tracker docs.

Commit:
Merge branch 'release/3.5' into add-tracker-docs

russorat commented:
@awholegunch  all green

Commit:
updating screenshot

Commit:
Edited RESTful API and split into its own file.
Fix tabbed-parsed-literal to correctly handle both JSON objects and
arrays, and fixed an error in ending the JSON data.

Commit:
Merge branch 'add-tracker-docs' of https://github.com/caskdata/cdap into add-tracker-docs

awholegunch commented:
LGTM

russorat commented:
LGTM
